earthpy/tests/test_plot.py

```diff
@@ -28,27 +27,27 @@ def test_num_titles():
     should raise an error OR if the title list is a different length than
     the array it should also raise an errors"""
 
-    single_band = im[0]
+    single_band = im_arr[0]
 
     with pytest.raises(ValueError):
         es.plot_bands(arr=single_band,
                       title=["Title1", "Title2"])
     with pytest.raises(ValueError):
-        es.plot_bands(arr=im,
+        es.plot_bands(arr=im_arr,
                       title=["Title1", "Title2", "Title3"])
 
 def test_num_axes():
```

Comment:
Same comment here - the test uses a module-level data object that could be a fixture: 

```python
def test_num_axes(image_array_3d):
    """If provided with a 2 band array, plot_bands should return 3 axes by
    default"""
    fig, ax = es.plot_bands(image_array_3d)
    assert len(fig.axes) == 3
```

Reply:
i'll be honest... i'm slightly afraid of putting fig objects in fixtures right now. i haven't done it so far. i want to be able to explicetly close them and know i can from a function but feel a bit bettter with them outside of the function. Particularly because i want to grab different parts of the object for diff tests. see what you think when i push these changes. i've also had problems trying to send TWO fixtures to a single test. in this case i'd want the array and the figure fixture. anyway see what you think when i push.  i'm open to whatever is the best approach.

Reply:
Word - I was suggesting that the array is passed as a fixture, rather than the figure. I'm not sure how passing figures as fixtures would work, but the changes you made to use fixtures for numpy arrays look great to me!